packages/core/admin/admin/src/components/DescriptionComponentRenderer.tsx

```diff
@@ -17,74 +20,76 @@ interface DescriptionComponent<Props, Description> {
  * DescriptionComponentRenderer
  * -----------------------------------------------------------------------------------------------*/
 
-/**
- * A collection lets us compositionally render components that can inject data into "slots"
- * that are sent to a Context Provider meaning we can access this data via `useCollection`.
- *
- * A common use case could be if you're creating a select component, you could render your options
- * like:
- * ```tsx
- * return (
- *  <Select>
- *    <Option>option 1</Option>
- *    <Option>option 2</Option>
- *  </Select>
- * )
- * ```
- * and the parent `Select` component would be able to access the Option data. In this case,
- * we render the descriptions and access them above to then pass to UI to render these data objects.
- */
-const [Collection, useCollection] = createCollection<never, any>('DescriptionData');
-
 interface DescriptionComponentRendererProps<Props = any, Description = any> {
-  children: (descriptions: Description[]) => React.ReactNode;
+  children: (descriptions: Array<Description & { id: string }>) => React.ReactNode;
   descriptions: DescriptionComponent<Props, Description>[];
   props: Props;
 }
 
-const DescriptionComponentRendererImpl = <Props, Description>({
+const DescriptionComponentRenderer = <Props, Description>({
   children,
   props,
   descriptions,
 }: DescriptionComponentRendererProps<Props, Description>) => {
-  const [items, setItems] = React.useState<Description[]>([]);
-  const { subscribe } = useCollection(undefined);
+  const statesRef = React.useRef<Record<string, { value: Description & { id: string } }>>({});
+  const [tick, forceUpdate] = useForceUpdate();
+
+  const requestHandle = React.useRef<number | null>(null);
+  const requestUpdate = React.useCallback(() => {
+    if (requestHandle.current) {
+      cancelIdleCallback(requestHandle.current);
+    }
+
+    requestHandle.current = requestIdleCallback(() => {
+      requestHandle.current = null;
 
-  React.useEffect(() => {
-    const unsub = subscribe((state) => {
-      setItems(state);
+      forceUpdate();
     });
+  }, [forceUpdate]);
+
+  const throttledRequestUpdate = useThrottledCallback(requestUpdate, 60, { trailing: true });
+
+  const update = React.useCallback<DescriptionProps<Props, Description>['update']>(
+    (id, description) => {
+      if (description === null) {
+        delete statesRef.current[id];
+      } else {
+        const current = statesRef.current[id];
+        statesRef.current[id] = { ...current, value: { ...description, id } };
```

Comment:
Why not mutate `statesRef.current[id].value` directly?

Reply:
FP over OOP i'd say :)